### ISSUE TITLE
Solution thread is missing its timing deadline

### ISSUE THREAD

**Issue by fnoble** [swift-nav/piksi_firmware#141]
The `time_matched_obs` thread is currently holding the `base_obss_lock` for a long time, causing the solution thread to very occasionally miss its timing deadline.

Moving `process_matched_obs()` out of the lock zone improves system liveness considerably and fixes an issue where the solution thread will miss its timing deadline. However, it would introduce an (unlikely) possibility that the low latency baseline could be calculated when the filter is in the process of being updated.

The proposed fix would be to have the low latency baseline calculation use a local copy of the ambiguity state that is updated with a much cheaper call that just copies the state out of the filter. This update would be called every time directly after `dgnss_update()`. The lock zone could then be just around this update call.

**Comment by fnoble** [comment]
Design snippet:

**Why:** As described in the issue the `time_matched_obs` thread is holding the `base_obss` lock way to long. If the `solution` thread waits on this lock then `time_matched_obs` thread inherits a super high priority and everyone else gets screwed. This is causing all kinds of timing deadlines to get missed in the system.

**How:** One lock is currently locking `base_obss` and serving a sneaky second purpose of locking the filter state so that we don't calculate a baseline half way through modifying something. We need to split this up into two locks. First we move the current lock so it only locks the call to `single_diff`. Next we refactor some things in `dgnss_management` so we can pull out the current ambiguity set after each update. We also refactor the low latency baseline calculation to use this state that is external to the filter. We can now lock that external state only which is only accessed for a short period of time.

**Testing:** Will add unit tests as I touch things in libswiftnav (yay). Then need to test on HITL due to rare occurrence of some of these subtle concurrency issues.

**Comment by fnoble** [comment]
Sprint 6/8 completion criteria: Tested and integrated
